### ISSUE TITLE
Origo node v0.5.0 mining stops every 2 hours

### ISSUE THREAD

**Issue by Shekelme** [origolab/origo-binary#14]
I have origo node deployed on HiveOS and the miner is pointed to this node. Every 2 hours (a little more actually) the mining stops. The node itself looks to be running, but there are 0 Sol/s until reboot. Why is this happening and how to fix this?

Here are these periods on temperature graph:

![image](https://user-images.githubusercontent.com/26449500/74589742-88b35200-503a-11ea-9474-a3b1be79d7b2.png)

On the miner side:
![image](https://user-images.githubusercontent.com/26449500/74595258-03e42a80-5072-11ea-8b62-a78353dbb189.png)

On the node side:
![image](https://user-images.githubusercontent.com/26449500/74595267-1e1e0880-5072-11ea-9819-4360788ebb95.png)

**Comment by Shekelme** [comment]
After miner restart mining process starts normally. Looks like the problem is in miner.

**Comment by ericripi** [comment]
Can you help us to get the log of the miner， to start the miner with log，here is an example:
```
./ogominer --use 0,1,2 -v -v -v > log.log
```
post the log to us when the miner has stopped working.

sometimes the miner will crash down when meeting some hardware errors, we suggest that you run a monitor to restart the miner when the error happens, following is an example of monitor.sh, 
```
#!/bin/sh  

proc_name="ogosolver"      
  
proc_num()                 
{
    num=`ps -ef | grep $proc_name | grep -v grep | wc -l`
    return $num
}

proc_num  
number=$?                  
if [ $number -ne 6 ]       
then                       
    service ogominer stop
	sleep 10
	service ogominer start
fi  
```

**Comment by Shekelme** [comment]
The log has been created but it is completely empty.

**Comment by ericripi** [comment]
I have noticed that your miner's state is not correct.  the speed is 0 sol/s,  there should be some error when the miner startup,  does the driver installed correctly？ you can check it with our guide [Ogo mining guide](https://github.com/origolab/origo-binary/blob/master/docs/gpu_mining_tutorial.md)

**Comment by Shekelme** [comment]
The speed at miner start is 13 Sol/s, this state when it shows 0 Sol/s is my problem that begins after a while (may be 1 hour, may be 2 hours).

**Comment by ericripi** [comment]
we have tested the miner with the card of `GTX 1070 / GTX 1070 Ti / GTX 1080 / GTX 1080 Ti`,  which type of card are you mining on?

please check if the process of  `ogosolver` exists when the problem happened,  the command of 
`ps aux | grep ogosolver`  can give us the state of the process.  if the process doesn't exist,  we should restart the miner.

**Comment by Shekelme** [comment]
I use this miner on single Radeon VII GPU. Latest stable HiveOS 0.6-119@200219.

Here is an output of ps aux | grep ogosolver
![image](https://user-images.githubusercontent.com/26449500/75009393-27153c80-54ad-11ea-8b89-c57054b71cfb.png)
 And I can always resume screen session where the miner is still running, but it doesn't hashing, as I showed above.

**Comment by sniper-noob** [comment]
User issue.. no problem with the node or miner here:)

**Comment by Shekelme** [comment]
Excellent answer, lol. What causes this issue?

**Comment by sniper-noob** [comment]
Well- I don't see any issue so far with node and miner. So difficult to answer the Q you ask here.

**Comment by Shekelme** [comment]
So the word 'exception" in miner's output does not sounds suspicious for you?

**Comment by sniper-noob** [comment]
What I'm saying. Never seen this error😊

**Comment by Shekelme** [comment]
This does not mean that this error is absolutely impossible. Let developers review this issue and help me to fix it.

**Comment by sniper-noob** [comment]
If you have the issue so we must have the issue 2, since we use same software.. we dont have the issue. Good luck man.

**Comment by Shekelme** [comment]
Excellent support. That is how it should always be.

**Comment by sniper-noob** [comment]
Well I'm not part of dev. So not here to support. I'm just sharing my experience I have with the node and miner.

**Comment by Shekelme** [comment]
So thank you for your sharing. Really, I'm appreciate this so much.

**Comment by sniper-noob** [comment]
Dosent sound you do. Goodluck

**Comment by ericripi** [comment]
@Shekelme  we have not test the miner on Radeon VII, we will try to find a Radeon VII and test on it later.

**Comment by sniper-noob** [comment]
@ericripi if [ $number -ne 6 ]       what this 6 stand for? number of gpus?

**Comment by Shekelme** [comment]
> @Shekelme we have not test the miner on Radeon VII, we will try to find a Radeon VII and test on it later.

I can give an access to Hive shell to this host running Radeon VII if it help.

**Comment by ericripi** [comment]
> @ericripi if [ $number -ne 6 ] what this 6 stand for? number of gpus?

yes,  for example, there is six GPU card in your rig,  you can check if all the card work correctly

**Comment by Shekelme** [comment]
> I can give an access to Hive shell to this host running Radeon VII if it help.

So nobody needs my help, right?

**Comment by sniper-noob** [comment]
@Shekelme takes time to build kernels for new gpu I think.. dev wont have time With this Corona time. You have to wait.
